tests/next_tests/unit_tests/program_processor_tests/runners_tests/dace_tests/test_dace.py

```diff
@@ -8,54 +8,81 @@
 
 """Test specific features of DaCe backends."""
 
-import ctypes
 import unittest
-from typing import Any
 
 import numpy as np
 import pytest
 
+import gt4py._core.definitions as core_defs
 import gt4py.next as gtx
-from gt4py.next import int32
 from gt4py.next.ffront.fbuiltins import where
 
 from next_tests.integration_tests import cases
 from next_tests.integration_tests.cases import (
+    E2V,
     cartesian_case,
+    unstructured_case,
 )
 from next_tests.integration_tests.feature_tests.ffront_tests.ffront_test_utils import (
     exec_alloc_descriptor,
+    mesh_descriptor,
 )
+from unittest.mock import patch
 
 from . import pytestmark
 
 dace = pytest.importorskip("dace")
 
 
-def get_scalar_values_from_sdfg_args(
-    args: tuple[list[ctypes._SimpleCData], list[ctypes._SimpleCData]],
-) -> list[Any]:
-    runtime_args, init_args = args
-    return [
-        arg.value for arg in [*runtime_args, *init_args] if not isinstance(arg, ctypes.c_void_p)
-    ]
+def make_mocks(monkeypatch):
+    # Wrap `compiled_sdfg.CompiledSDFG.fast_call` with mock object
+    mock_fast_call = unittest.mock.MagicMock()
+    mock_fast_call_attr = dace.codegen.compiled_sdfg.CompiledSDFG.fast_call
+
+    def mocked_fast_call(self, *args, **kwargs):
+        mock_fast_call.__call__(*args, **kwargs)
+        fast_call_result = mock_fast_call_attr(self, *args, **kwargs)
+        # invalidate all scalar positional arguments to ensure that they are properly set
+        # next time the SDFG is executed before fast_call
+        positional_args = set(self.sdfg.arg_names)
+        sdfg_arglist = self.sdfg.arglist()
+        for i, (arg_name, arg_type) in enumerate(sdfg_arglist.items()):
+            if arg_name in positional_args and isinstance(arg_type, dace.data.Scalar):
+                self._lastargs[0][i] = None
+        return fast_call_result
+
+    monkeypatch.setattr(dace.codegen.compiled_sdfg.CompiledSDFG, "fast_call", mocked_fast_call)
+
+    # Wrap `compiled_sdfg.CompiledSDFG._construct_args` with mock object
+    mock_construct_args = unittest.mock.MagicMock()
+    mock_construct_args_attr = dace.codegen.compiled_sdfg.CompiledSDFG._construct_args
```

Comment:
This is fully optional, but I think the name is a bit misleading, I would suggest `constrcut_args_impl` or something, because the prefix `mock` implies that it is also in someway decorated or so.
But feel free to ignore.